e2e/app/page/google-login.ts

```diff
@@ -118,32 +114,34 @@ export default class GoogleLoginPage extends BasePage {
         console.error('Google login button not found. ' + err);
         throw err;
       });
-      await this.clickAndWait(googleLoginButton);
+      await BaseElement.asBaseElement(this._page, googleLoginButton).clickAndWait();
 
       if (!user || user.trim().length === 0) {
         console.warn('Login user email: value is empty!!!')
       }
       await this.enterEmail(user);
+      await this._page.waitFor(1000); // reduce chances of getting Google login recaptcha
       await this.enterPassword(pwd);
       await this.submit();
     } catch (err) {
-      await takeScreenshot(this.page, 'FailedLoginPage');
-      await savePageToFile(this.page, 'FailedLoginPage');
-      throw err;
+      await takeScreenshot(this._page);
```

Comment:
This code is duplicated from what we saw before. Maybe make a common method or class you can use for cleaning up failed tests and reporting on them consistently.